src/app/static/src/app/store/load/actions.ts

```diff
@@ -78,8 +96,8 @@ async function getFilesAndLoad(context: ActionContext<LoadState, RootState>, fil
         .withError("LoadFailed")
         .postAndReturn<Dict<LocalSessionFile>>("/session/files/", files)
         .then(() => {
-            if (state.loadingState == LoadingState.UpdatingState) {
-                dispatch("updateStoreState", savedState);
+            if (state.loadingState != LoadingState.LoadFailed) {
+                dispatch("load/updateStoreState", savedState, {root: true});
```

Comment:
I changed this from a local dispatch because I wanted to write a meaningful integration test which tested the load action actually creating the project in the projects module via the dispatch, so I didn't want to just mock the dispatch. So I create a store in the integration test and take pass its dispatch to the action, but I couldn't work out how to make this a localised dispatch. Any ideas?

Reply:
2 options:
1. use a fake local dispatch in the test:
```
const localDispatch = (name: string, payload: any, options: DispatchOptions) => {
            if (!options) {
                return store.dispatch("load/" + name, payload)
            } else {
                return store.dispatch(name, payload, options)
            }
}
```
2. use the actual module's dispatch:
```
const moduleDispatch = ((store as any)._modulesNamespaceMap["load/"] as any).context.dispatch
```

Reply:
I guess for an accurate integration test the latter is better, even though you have to access private methods and cast everything to `any`.

Reply:
Agreed, thanks.